Date linting updates

paulsturgess opened the pull request:
To prepare for the future upgrade of [carbon-factory](https://github.com/Sage/carbon-factory), that will introduce stricter linting rules, this PR updates the `date` components to pre-emptively resolve the errors.

Updated components:
- Date
- DateRange

#### Functional changes
After QA feedback I have updated the props of the input to set disabled to true on the input if either `disabled` or `readOnly` are set to true by the developer.

**BEFORE**
![date-disabled-before](https://cloud.githubusercontent.com/assets/4964/26777787/6c2d34b0-49d6-11e7-87f5-a038e9ce6390.gif)

**AFTER**
![date-disabled-after](https://cloud.githubusercontent.com/assets/4964/26777794/70c53aea-49d6-11e7-82fe-21211aedf115.gif)

ianoxley commented:
Found an issue during QA where the `disabled` and `readOnly` props on `Date` don't have the desired effect.

paulsturgess commented:
@ianoxley I've updated this PR with an update around the disabled/readonly behaviour. I've also added a couple of gifs to the description to show the change.

@abramin this is the commit that ensures the input can be set as disabled: https://github.com/Sage/carbon/pull/1284/commits/007db95fbea46970791c0a2fd08e029c6101784f

abramin commented:
👍 Will re-QA to double check

ianoxley commented on src/components/date/__spec__.js:
```diff
@@ -381,6 +385,10 @@ describe('Date', () => {
       });
+
+      it('sets the input as disabled', () => {
+        expect(instance._input.disabled).toEqual(true);
```
Should this be testing for the `readonly` attribute instead of `disabled`?

paulsturgess replied:
No an input has a native html attribute of `disabled`. Both props (`disabled` and `readOnly`) should effect this attribute.

ianoxley replied:
Ah, but `input` elements have a `readonly` attribute too: https://developer.mozilla.org/en/docs/Web/HTML/Element/input#attr-readonly

So I think the `readOnly` prop should set the `readonly` attribute.

paulsturgess replied:
Ha ok I didn't know that! Good catch, will update.

paulsturgess commented:
@ianoxley I've updated so that the `readOnly` prop now sets the `readonly` attribute on the input.

sianford commented:
Retested in Safari, Chrome and Firefox browsers. The disabled and readOnly props now work as expected. Tests passed.
